packages/client/src/pages/page-editor/page-editor.vue

```diff
@@ -102,365 +101,343 @@ import { collectPageVars } from '@/scripts/collect-page-vars';
-				items: [
-					{ value: 'if', text: this.$ts._pages.blocks.if },
-					{ value: 'post', text: this.$ts._pages.blocks.post }
-				]
-			}];
-		},
-
-		getScriptBlockList(type: string = null) {
-			const list = [];
-
-			const blocks = blockDefs.filter(block => type === null || block.out === null || block.out === type || typeof block.out === 'number');
-
-			for (const block of blocks) {
-				const category = list.find(x => x.category === block.category);
-				if (category) {
-					category.items.push({
-						value: block.type,
-						text: this.$t(`_pages.script.blocks.${block.type}`)
-					});
-				} else {
-					list.push({
-						category: block.category,
-						label: this.$t(`_pages.script.categories.${block.category}`),
-						items: [{
-							value: block.type,
-							text: this.$t(`_pages.script.blocks.${block.type}`)
-						}]
-					});
-				}
-			}
+	varName = varName!.trim();
 
-			const userFns = this.variables.filter(x => x.type === 'fn');
-			if (userFns.length > 0) {
-				list.unshift({
-					label: this.$t(`_pages.script.categories.fn`),
-					items: userFns.map(v => ({
-						value: 'fn:' + v.name,
-						text: v.name
-					}))
-				});
-			}
+	if (hpml.isUsedName(varName)) {
+		os.alert({
+			type: 'error',
+			text: i18n.ts._pages.variableNameIsAlreadyUsed
+		});
+		return;
+	}
 
-			return list;
-		},
+	const id = uuid();
+	variables.push({ id, name: varName, type: null });
+}
 
-		setEyeCatchingImage(e) {
-			selectFile(e.currentTarget ?? e.target, null).then(file => {
-				this.eyeCatchingImageId = file.id;
-			});
-		},
+function removeVariable(v) {
+	variables = variables.filter(x => x.name !== v.name);
+}
+
+function getPageBlockList() {
+	return [{
+		label: i18n.ts._pages.contentBlocks,
+		items: [
+			{ value: 'section', text: i18n.ts._pages.blocks.section },
+			{ value: 'text', text: i18n.ts._pages.blocks.text },
+			{ value: 'image', text: i18n.ts._pages.blocks.image },
+			{ value: 'textarea', text: i18n.ts._pages.blocks.textarea },
+			{ value: 'note', text: i18n.ts._pages.blocks.note },
+			{ value: 'canvas', text: i18n.ts._pages.blocks.canvas },
+		]
+	}, {
+		label: i18n.ts._pages.inputBlocks,
+		items: [
+			{ value: 'button', text: i18n.ts._pages.blocks.button },
+			{ value: 'radioButton', text: i18n.ts._pages.blocks.radioButton },
+			{ value: 'textInput', text: i18n.ts._pages.blocks.textInput },
+			{ value: 'textareaInput', text: i18n.ts._pages.blocks.textareaInput },
+			{ value: 'numberInput', text: i18n.ts._pages.blocks.numberInput },
+			{ value: 'switch', text: i18n.ts._pages.blocks.switch },
+			{ value: 'counter', text: i18n.ts._pages.blocks.counter }
+		]
+	}, {
+		label: i18n.ts._pages.specialBlocks,
+		items: [
+			{ value: 'if', text: i18n.ts._pages.blocks.if },
+			{ value: 'post', text: i18n.ts._pages.blocks.post }
+		]
+	}];
+}
 
-		removeEyeCatchingImage() {
-			this.eyeCatchingImageId = null;
-		},
+function getScriptBlockList(type: string = '') {
```

Comment:
I think adjusting the default value is not the right choice here, because how it is usedin `page-editor.script-block.vue` might actually pass in `null`.
```suggestion
function getScriptBlockList(type: string | null = null) {
```